### PR TITLE
Introduce billing meters

### PR DESCRIPTION
**Pull request by byucesoy**
Rest of the usage record logic depends on product naming and requires some modification on the DB side, which will be integrated with separate commit. I'm isolating some independent part of it as this commit.

This commit introduces billing meters, which will be used as catalog for all pricing calculations and invoices. Each place that needs to calculate a price will be able to fetch unit price of particular resource from this table, and then calculate the total price as unit_price * usage_amount * usage_duration.

For example, for VMs, if the a customer uses 4 cores from c5a family for 1000 minutes in hetzner-fsn1 region, total price would be 0.000171296 * 4 * 1000. Unit of unit_price for VMs is minute, but each product can determine its own unit. For disks, it could be GB or for a serverless product it could be number of requests. It is possible to add human readable unit column so eliminate potential confusion, but I'm leaving that out for now. We can add if we see a need for that.

Initially, I started to build more general solution that would satisfy the needs of many future products, but that proved itself quite challenging as there are lots of unknown unknowns. Instead, I decided to build much more simple and flexible model that can be extended as we learn more about the product we are building.

Current model is quite flexible and allows numerous potential expansion paths. For example, reserved instance pricing can be implemented easily by adding a column.

Similarly price updates can be done quite easily by adding two new columns; effective_since and effective_until. Then it is matter of ensuring invoice generation and price calculator honours those dates. Alternative would be generating new billing meters and depreciating the old ones, but updating billing meter references everywhere requires manual database surgery and is open to operator mistakes.

### CODE REVIEW

File: migrate/20230725_billing_meters.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    create_table(:billing_meter) do
+      column :id, :uuid, primary_key: true, default: nil
+      column :resource_type, :text, collate: '"C"', null: false
+      column :resouce_family, :text, collate: '"C"', null: false
+      column :location, :text, collate: '"C"', null: false
+      column :unit_price, :numeric, null: false
+      index [:resource_type, :resouce_family, :location], unique: true
+    end
+
+    # These are valid UBIDs for TYPE_BILLING_METER
```
Comment (fdr) [inline migrate/20230725_billing_meters.rb:14]: I think it should be okay to require in the UBID code into migrations, though we should avoid dependencies on models per normal.

File: migrate/20230725_billing_meters.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+Sequel.migration do
+  up do
+    create_table(:billing_meter) do
```
Comment (fdr) [inline migrate/20230725_billing_meters.rb:5]: Isn't it more of a `billing_rate`?

**Review by fdr — CHANGES_REQUESTED** [review]
Hard to get too worked up about thirty lines of code, most in a pretty simple migration...but the name doesn't seem right to me? Perhaps I'm wrong and need to spend more time with the dictionary.

File: migrate/20230725_billing_meters.rb (hunk 2 above)
Comment (byucesoy) [reply migrate/20230725_billing_meters.rb]: Naming comes from [Azure](https://learn.microsoft.com/en-us/rest/api/containerapps/stable/billing-meters/get?tabs=HTTP#billingmeter), and it seems it is used in [other places](https://www.moesif.com/docs/guides/guide-on-creating-a-billing-meter-with-stripe/) as well, but I don't have strong attachment to this name. `billing_rate` is also good and possibly more meaningful to people who are not familiar with Azure.

File: migrate/20230725_billing_meters.rb (hunk 1 above)
Comment (byucesoy) [reply migrate/20230725_billing_meters.rb]: Not very happy to put these here, but these ids needs to be generated offline and somehow put into to database, so migrations seemed like an OK place to do that.

Comment (byucesoy) [reply migrate/20230725_billing_meters.rb]: Well... they don't need to be valid UBIDs I guess.

**Review by fdr — APPROVED** [review]

**Review by enescakir — APPROVED** [review]

File: migrate/20230725_billing_meters.rb (hunk 1 above)
Comment (byucesoy) [reply migrate/20230725_billing_meters.rb]: I decided to keep them valid UBIDs for now. If we want, ignoring their UBIDness is easy but making non UBID uuids to UBID is more difficult.

**Pull request merged**